### PR TITLE
[CRIMAP-715] Remove BaseDecorator in favour of BasePresenter

### PR DESCRIPTION
**Pull request by MohammedHassanLatif**
## Description of change
Refactored code changed from BaseDecorator to BasePresenter. Removed other decorator code and files.

## Link to relevant ticket
https://dsdmoj.atlassian.net/jira/software/projects/CRIMAP/boards/897?selectedIssue=CRIMAP-715

### CODE REVIEW

File: app/decorators/address_form_decorator.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class AddressFormDecorator < BaseDecorator
+class AddressFormDecorator < BasePresenter
```
Comment (willmcb) [inline app/decorators/address_form_decorator.rb:1]: Why not make this a `AddressFormPresenter` as well that way we aren't mixing concepts

File: app/presenters/crime_application_presenter.rb [hunk 2]
```diff
@@ -14,6 +14,8 @@ def applicant_dob
   end
 
   def applicant_name
+    return unless applicant?
```
Comment (willmcb) [inline app/presenters/crime_application_presenter.rb:17]: Is this a boy scout style addition unrelated to the general refactor you're doing here in relation to removing the concept of Decorators?

**Review by willmcb — CHANGES_REQUESTED** [review]
Few minor things to think about - looks good though happy to merge once they have been looked at / we have had a chat.

**Review by hiboabd — APPROVED** [review]
Looks good to me! 👏

File: app/presenters/crime_application_presenter.rb (hunk 2 above)
Comment (willmcb) [inline app/presenters/crime_application_presenter.rb]: This can be removed / put back to its orignal state

**Review by willmcb — APPROVED** [review]
Looks good :)

**Pull request merged**